packages/sdk/src/clients/multicall/client.ts

```diff
@@ -1,8 +1,8 @@
 import { EthersContracts } from "@uma/core";
 import type { SignerOrProvider } from "../..";
 
-export type Instance = EthersContracts.Multicall;
-const Factory = EthersContracts.Multicall__factory;
+export type Instance = EthersContracts.Multicall2;
+const Factory = EthersContracts.Multicall2__factory;
```

Comment:
i actually think your initial feeling here on creating a separate client is right. i think we should keep multicall 1 and 2 distinct

Reply:
and then `class Multicall`'s State would look like
```
interface State {
  requests: Request[];
  multicallClient: multicall.Instance | multicall2.Instance;
}
```
also `class Multicall` would have an additional method 
`public async smth_like_read_with_errors(_requests: Request[] = this.requests)` that will throw an exception if multicall client is not version 2

what do you think?

Reply:
Yea I think that we can also create a `class Multicall2` as a separate file as well which extends from `class Multicall` and just adds the new function.  I can write up a new card for this and we can also discuss more tomorrow.